### PR TITLE
Issue #563 More flexible way of node renaming

### PR DESCRIPTION
**Pull request by stepanzharychevbroadcom**
## Proposed changes

This branch adds more flexible way of node renaming. It includes verification file status and allows to save it before rename will happen, so your changes won't be lost.

Milestone: **1.7**

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] `npm run vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zdmullen** [comment]
Also covers issue #567, it appears.

File: __tests__/__unit__/uss/USSTree.unit.test.ts [hunk 1]
```diff
@@ -14,12 +14,18 @@ import { createUSSTree, USSTree } from "../../../src/uss/USSTree";
 import { ZoweUSSNode } from "../../../src/uss/ZoweUSSNode";
```
Comment (VitGottwald) [inline __tests__/__unit__/uss/USSTree.unit.test.ts:14]: It looks like all changes in this file are related to formatting. Perhaps it is time, that we implement prettier https://github.com/zowe/vscode-extension-for-zowe/issues/815

**Comment by VitGottwald** [comment]
@stepanzharychevbroadcom could you look at the issues @katelynienaber found out above?

**Comment by stepanzharychevbroadcom** [comment]
@zdmullen @katelynienaber I fixed first issue, but 2 others will need the separate ticket. Problem there is deeper than this branch, logic of USS favorite adding ignores current search pattern. So if you operate under `/tmp/test-folder` pattern and add `test.txt` file to favourites it's full path will be saved as `/test.txt` which is incorrect and causes error, so it should be reworked.

**Comment by katelynienaber** [comment]
@stepanzharychevbroadcom Thank you ^^ The first issue looks fine now. Could you link to the issue created for other 2 items? I don't want to lose them :) I will approve after this.

**Comment by katelynienaber** [comment]
Posting link to opened issue for #2/#3 bugs I found above: [#911](https://github.com/zowe/vscode-extension-for-zowe/issues/911)

**Comment by zFernand0** [comment]
> # [Codecov](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=h1) Report
> > Merging [#817](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=desc) into [master](https://codecov.io/gh/zowe/vscode-extension-for-zowe/commit/135893d0415a70c02e6cd6064836771275cf9240&el=desc) will **decrease** coverage by `1.26%`.
> > The diff coverage is `66.96%`.
> 
> [Impacted file tree graph](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=tree)
> 
> ```diff
> @@            Coverage Diff             @@
> ##           master     #817      +/-   ##
> ==========================================
> - Coverage   90.78%   89.51%   -1.27%     
> ==========================================
>   Files          49       51       +2     
>   Lines        4364     4494     +130     
>   Branches      767      791      +24     
> ==========================================
> + Hits         3962     4023      +61     
> - Misses        398      467      +69     
>   Partials        4        4              
> ```
> 
> [Impacted Files](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=tree)	Coverage Δ	
> [src/config/constants.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL2NvbmZpZy9jb25zdGFudHMudHM=)	`100.00% <ø> (ø)`	
> [src/uss/actions.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL3Vzcy9hY3Rpb25zLnRz)	`76.47% <29.41%> (-1.24%)`	⬇️
> [src/dataset/actions.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL2RhdGFzZXQvYWN0aW9ucy50cw==)	`86.04% <30.00%> (-8.70%)`	⬇️
> [src/dataset/DatasetTree.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL2RhdGFzZXQvRGF0YXNldFRyZWUudHM=)	`81.77% <66.66%> (-0.33%)`	⬇️
> [src/extension.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL2V4dGVuc2lvbi50cw==)	`85.77% <68.75%> (-1.26%)`	⬇️
> [src/utils/workspace.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3dvcmtzcGFjZS50cw==)	`79.62% <79.16%> (ø)`	
> [src/uss/ZoweUSSNode.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL3Vzcy9ab3dlVVNTTm9kZS50cw==)	`84.78% <90.90%> (-0.04%)`	⬇️
> [src/uss/USSTree.ts](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree#diff-c3JjL3Vzcy9VU1NUcmVlLnRz)	`88.23% <100.00%> (+0.08%)`	⬆️
> ... and [1 more](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817/diff?src=pr&el=tree-more)		
> [Continue to review full report at Codecov](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=continue).
> 
> > **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> > `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> > Powered by [Codecov](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=footer). Last update [c59bf21...e12823b](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

Is this for real?
When you click the [`Continue to review full report at Codecov.`](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=continue), it just says 91.07%

**Comment by zFernand0** [comment]
> > Merging [#817](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=desc) into [master](https://codecov.io/gh/zowe/vscode-extension-for-zowe/commit/135893d0415a70c02e6cd6064836771275cf9240&el=desc) will **decrease** coverage by `1.26%`.
> > The diff coverage is `66.96%`.
> > 
> 
> Is this for real?
> When you click the [`Continue to review full report at Codecov.`](https://codecov.io/gh/zowe/vscode-extension-for-zowe/pull/817?src=pr&el=continue), it just says 91.07%

Nvm, it's all good now!

**Comment by zFernand0** [comment]
@stepanzharychevbroadcom next time, please use the complete the template 😋

**Pull request merged**